### DISCUSSION TITLE
If using CFG++ samplers, now CFG from 0 and onwards works, including CFG 1.

### DISCUSSION THREAD

**Discussion by Panchovix** [Panchovix/stable-diffusion-webui-reForge#186] [Announcements]
Hi there, hope you guys are fine.

After some tests, CFG++ samplers now on reForge work with CFG with values from 0 and onwards, and not like in the past like 1 disabled it and 0.01 to 0.99 reversed the negative with the positive prompts.

 Note, this applies only for CFG++ samplers. For normal samplers, CFG 1 is disabled and below this it reverses the neg with pos and viceversa.,

If you want to be able to set CFG below 1, you have to edit the `ui-config.json` file that gets **generated after launching the webui for the first time**.

Here, search for "txt2img/CFG Scale/minimum" (or "img2img/CFG Scale/minimum" for img2img) and change the value from 1.0 to 0.0 or 0.1, or 0.001 etc.

You can set the stepping on "txt2img/CFG Scale/step", default is 0.5.

Hope this helps for the users that are using CFG++ samplers.

Take care!

**Comment by ThisIsNetsu** [comment]
I just wanted so say thank you. I expected reForge to go completely dormant, now that Forge also gets updates again. But you keep updating it. I stick with reForge because you have SO many more samplers and schedulers than forge.
I really do appreciate all the work you put into this!

**Comment by Reithan** [comment]
I could do without some of the niche samplers and schedules, but the fact that reForge is still so much faster than Forge on my machine is a game-changer!

Thanks for all your work!
